core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorSidebarDirective.js

```diff
@@ -16,17 +16,155 @@
+        };
+        // TODO: scroll to the nav link in case not displayed
+      }
+    },
+    scrollTo: function(hash) {
+      $anchorScroll.yOffset = 80;
+      $anchorScroll(hash);
+    }
+  };
+
+  // Onscroll event handler
+  $document.on('scroll', function() {
+    var documentY = $document.scrollTop();
+    var positiveTargets = {};
+    var positiveTargetsDy = [];
+    // Find the closest positive target
+    for (var i = 0, targetsLength = scrollSyncService._targets.length;
+          i < targetsLength; i++) {
+      var target = angular.element(scrollSyncService._targets[i]);
+      var targetOffset = target.offset().top;
+      var dy = targetOffset - documentY;
+      if (dy > 0) {
+        positiveTargets[dy] = target;
+        positiveTargetsDy.push(dy);
+      }
+    };
+    for (var i = 0, targetsLength = positiveTargetsDy.length,
+          minPositive = null; i < targetsLength; i++) {
+      if (i === 0) {
+        minPositive = positiveTargetsDy[i];
+        continue;
+      } else {
+        var dy = positiveTargetsDy[i];
+        if (minPositive > dy) {
+          minPositive = dy;
+        }
+      }
+    };
+    if (minPositive === null) {
+      return;
+    } else {
+      // Update the sidebar active element
+      var closestTarget = positiveTargets[minPositive];
+      scrollSyncService.updateActive(closestTarget.attr('id'));
+    }
+  });
+
+  return scrollSyncService;
+}]);
+
+// Used to mark parts of the editor that have a sidebar item associated to it.
+oppia.directive('scrollSync', ['ScrollSyncService', function(ScrollSyncService) {
+  return {
+    restrict: 'A',
+    link: function(scope, element) {
+      ScrollSyncService.addTarget(element);
+    }
+  };
+}]);
+
 oppia.directive('simpleEditorSidebar', [function() {
   return {
     restrict: 'E',
     templateUrl: 'simpleEditor/sidebar',
     controller: [
-        '$scope', 'EditorModeService', 'SimpleEditorManagerService',
-        function($scope, EditorModeService, SimpleEditorManagerService) {
+       '$scope', 'EditorModeService', 'SimpleEditorManagerService',
+        'ScrollSyncService',
+        function($scope, EditorModeService, SimpleEditorManagerService,
+                 ScrollSyncService) {
       $scope.SUBFIELD_LABELS = [
-        'Multiple choice', 'Correct answer', 'Hints', 'Bridge text'];
+        {label: 'Multiple choice', abbr: 'mc'},
+        {label: 'Correct answer', abbr: 'ca'},
+        {label: 'Hints', abbr: 'hint'},
+        {label: 'Bridge text', abbr: 'bt'}
+      ];
       $scope.setEditorModeToFull = EditorModeService.setModeToFull;
-      $scope.questionList = SimpleEditorManagerService.getQuestionList();
+      $scope.questionList = ScrollSyncService._questionList = SimpleEditorManagerService.getQuestionList();
+      $scope.scrollTo = function(hash) {
+        ScrollSyncService.scrollTo(hash);
+      };
+      $scope.collapse = function(question) {
+        if (!angular.isDefined(question.collapsed)) {
```

Comment:
Why would it not be defined? Perhaps we should have this always be defined from the start? Don't think it's a good idea to introduce new properties midway through execution.